ocean_lib/web3_internal/web3_overrides/contract.py

```diff
@@ -118,8 +123,16 @@ def transact_with_contract_function(
         txn_hash = web3.eth.send_transaction(transact_transaction)
 
     network_id = get_chain_id(web3)
-    web3.eth.wait_for_transaction_receipt(
+    receipt = web3.eth.wait_for_transaction_receipt(
         txn_hash, get_network_timeout(network_id=network_id)
     )
+    if transaction["chainId"] == 1337:
```

Comment:
I am not very familiar with this functionality but it seems to check out. I like it. However, I would suggest, because there is a lot of code duplication, to isolate this if/else block + the wait_for_transaction_receipt part into a utils function that takes a web3, block number and transaction hash as a parameters. More parameters might be needed, but it would be useful to have a single entry point for this logic. And you'd have to import DEFAULT_BLOCK_CONFIRMATIONS in just one place (single responsibility principle). How does that sound?

Reply:
Sounds great. I'll do it. Thanks for the suggestion